### PR TITLE
Feat / react 18 compatibility

### PR DESCRIPTION
**Pull request by RCVZ**
## The problem

The React 18 update in the JW ott project caused the slider to show a slide animation while resetting the transform value after a slide. 

## The cause

In React 18 the state update has improved performance by batching more state updates. This caused the reset state and the after-reset state update to be batched as one causing the slide animation also to be visible when resetting the animation.

## The solution

Add the transition to the state so that we can update the transition together with the transform. And thereby having more control over the transition. There should also be a small performance improvement because the dom is now only updated by React.

**Note**
I also updated the dependencies.
Has also been tested in Symphony web and Symphony tizen(User-Agent).

### CODE REVIEW

File: src/TileSlider/TileSlider.tsx [hunk 1]
```diff
@@ -205,19 +220,6 @@ const TileSlider = <T extends unknown>({
     [minimalTouchMovement, slide, onSwipeStart, onSwipeEnd],
   );
 
-  useEffect(() => {
-    if (state.afterReset) {
-      if (frameRef.current) frameRef.current.style.transition = transitionBasis;
-      if (onSlideEnd) onSlideEnd();
```
Comment (royschut) [inline src/TileSlider/TileSlider.tsx:211]: You're not calling this method anywhere now.

**Review by royschut — APPROVED** [review]
Really nicely found!

Although I'm not sure if `onSlideEnd()` is still used, I think you should call it somewhere else, or otherwise deprecate it?

File: src/TileSlider/TileSlider.tsx (hunk 1 above)
Comment (RCVZ) [reply src/TileSlider/TileSlider.tsx:211]: Its still used in Symphony

Comment (royschut) [reply src/TileSlider/TileSlider.tsx:211]: So you'll fix it? 😛

**Review by ChristiaanScheermeijer — COMMENTED** [review]
Nice solution. We indeed have to re-add the `onSlideEnd` callback. We can add it directly after calling the transition reset.

**Review by ChristiaanScheermeijer — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -13,37 +13,38 @@
...
-    "typescript": "^4.2.4"
+    "rollup-plugin-postcss": "^4.0.2",
+    "rollup-plugin-typescript2": "^0.35.0",
+    "typescript": "^5.1.5"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0"
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
```
Comment (ChristiaanScheermeijer) [inline package.json:48]: Wait, the module is still usable on React 17, right? Should this be set to:

```suggestion
    "react": "^17.0.0 || ^18.0.0",
    "react-dom": "^17.0.0 || ^18.0.0"
```

**Pull request merged**